modules/Account/app/Repositories/PermissionRepository.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Modules\Account\app\Repositories;
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Http\Request;
+use Illuminate\Pagination\LengthAwarePaginator;
+use Illuminate\Support\Facades\Hash;
+use Modules\Account\app\Contracts\Repositories\RoleRepositoryInterface;
+use Modules\Account\app\Http\Resources\UserCollection;
+use Modules\Account\app\Http\Resources\UserResource;
+use Modules\Account\app\Models\Permission;
+use Modules\Account\app\Models\Role;
+use Modules\Account\app\Models\User;
+
+class PermissionRepository implements RoleRepositoryInterface
+{
+
+    public function getAllWithRoles(Request $request)
+    {
+        $query = $this->applyFilters($request);
+
+        return new UserCollection($query->get());
+    }
+
+    private function applyFilters($request)
+    {
+        $query = User::query();
+        $search = $request->input('search');
+
+        $this->applySearch($query, $search);
```

Comment:
I think , the return value from applySearch must save in variable and then send in next method . 
please double check this.

Reply:
no need to return it will save in query